src/Database/TypeMap.php

```diff
@@ -17,144 +17,61 @@
 namespace Cake\Database;
 
 /**
- * Implements default and single-use mappings for columns to their associated types
+ * Maps fields to types.
  */
 class TypeMap
 {
     /**
-     * Associative array with the default fields and the related types this query might contain.
-     *
-     * Used to avoid repetition when calling multiple functions inside this class that
-     * may require a custom type for a specific field.
-     *
-     * @var array<string, string>
-     */
-    protected array $_defaults = [];
```

Comment:
There must be a reason why the default types and additional/overriding types were maintained separately isn't it?

Reply:
I think the idea was default types could come from the tables and then users could override specific types as needed. That doesn't happen for aliased types, but we should make sure Query doesn't overwrite user-specified types for a field.

The "one-use" stuff must have been part of a different designs as I can't find any code based around that.